tests/test_wps_generic_zonal_stats.py

```diff
@@ -100,3 +111,42 @@ def test_geoserver_dem_wcs(self):
 
         geometry = shape(feature['geometry'])
         assert isinstance(type(geometry), type(MultiPolygon))
+
+
+    def test_geoserver_dem_wcs_categorized(self):
+        client = client_for(Service(processes=[ZonalStatisticsProcess(), ], cfgfiles=CFG_FILE))
+        fields = [
+            'select_all_touching={touches}',
+            'categorical={categorical}',
+            'band={band}',
+            'shape=file@xlink:href=file://{shape}', ]
+
+        datainputs = ';'.join(fields).format(
+            touches=True,
+            categorical=True,
+            band=1,
+            shape=TESTDATA['mrc_subset'],
+        )
+        resp = client.get(
+            service='WPS', request='Execute', version='1.0.0', identifier='zonal-stats', datainputs=datainputs)
+
+        assert_response_success(resp)
+        out = get_output(resp.xml)
+        feature = json.loads(out['statistics'])['features'][0]
+
+        stats = feature['properties']
+        assert {'count', 'min', 'max', 'mean', 'median', 'sum', 'nodata'}.issubset(stats)
+
+        # Check for accurate pixel counts
```

Comment:
This piece of code seems to be a copy-paste from above. I suggest making it into a function.

Reply:
Sounds good. I can make it a testing function under `common`.